rpcserver.go

```diff
@@ -8067,27 +8069,36 @@ func (r *rpcServer) computeChannelAssetBalance(
 				"data: %w", err)
 		}
 
-		for assetIdx := range assetData.Assets {
-			assetOutput := assetData.Assets[assetIdx]
-			assetIDStr := assetOutput.AssetInfo.AssetGenesis.AssetID
-			assetIDBytes, err := hex.DecodeString(assetIDStr)
-			if err != nil {
-				return nil, fmt.Errorf("error decoding asset "+
-					"ID: %w", err)
-			}
-			var assetID asset.ID
-			copy(assetID[:], assetIDBytes)
+		// Check if the assets of this channel match the provided
+		// specifier.
+		pass, err := r.cfg.RfqManager.ChannelCompatible(
+			ctx, assetData.Assets, specifier,
+		)
+		if err != nil {
+			return nil, err
+		}
 
-			channelsByID[assetID] = append(
-				channelsByID[assetID], channelWithAsset{
-					assetInfo:   assetOutput,
-					channelInfo: openChan,
-				},
-			)
+		// Since the assets of the channel passed the above filter,
+		// we're safe to aggregate their info to be represented as a
+		// single entity.
+		var aggrAssetInfo rfqmsg.JsonAssetChanInfo
+
+		for _, assetInfo := range assetData.Assets {
```

Comment:
The whole JSON structure is very weird at the moment and I'm refactoring this to be correct in https://github.com/lightninglabs/taproot-assets/pull/1413. Can you please add a TODO here that we fix this?